### PR TITLE
bazel: add build defs for debian packages

### PR DESCRIPTION
**Pull request by mikedanese**

### CODE REVIEW

File: defs/gcs_uploader [hunk 1]
```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
```
Comment (spxtr) [inline defs/gcs_uploader]: `#!/usr/bin/env python`

File: defs/deb.bzl [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  pkg_deb(
+      name = name,
+      architecture = "amd64",
+      data = name + "-data",
+      depends = depends,
+      description = description,
+      homepage = KUBERNETES_HOMEPAGE,
+      maintainer = KUBERNETES_AUTHORS,
+      package =  name,
+      version = "1.6.0-alpha",
```
Comment (spxtr) [inline defs/deb.bzl]: How will this get set?

File: defs/gcs_uploader (hunk 1 above)
Comment (mikedanese) [inline defs/gcs_uploader]: Done

File: defs/deb.bzl (hunk 2 above)
Comment (mikedanese) [inline defs/deb.bzl]: I need to implement stamping, then it should work. It's not super urgent yet.

File: defs/gcs_uploader.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+import argparse
+import atexit
+import os
+import os.path
+import shutil
+import subprocess
+import sys
+import tempfile
+
+parser = argparse.ArgumentParser(description='Upload build targets to GCS.')
```
Comment (spxtr) [inline defs/gcs_uploader.py]: two newlines between top-level blocks

File: defs/gcs_uploader.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+parser.add_argument("--manifest",required=True, help="path to manifest of targets")
```
Comment (spxtr) [inline defs/gcs_uploader.py]: space before required

File: defs/gcs_uploader.py [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+parser.add_argument("--root",required=True, help="path to root of workspace")
+parser.add_argument("gcs_path", help="path in gcs to push targets")
```
Comment (spxtr) [inline defs/gcs_uploader.py]: required?

File: defs/gcs_uploader.py [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+def main(argv):
+  scratch = tempfile.mkdtemp(prefix="bazel-gcs.")
```
Comment (spxtr) [inline defs/gcs_uploader.py]: four spaces in the real world

File: defs/gcs_uploader.py [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+  atexit.register(lambda: shutil.rmtree(scratch))
+
+  with open(argv.manifest) as manifest:
+    for artifact in manifest:
+      artifact = artifact.strip()
+      try:
+        os.makedirs(os.path.join(scratch, os.path.dirname(artifact)))
+      except (OSError):
+        pass
```
Comment (spxtr) [inline defs/gcs_uploader.py]: is this check to not break for duplicate directories?

File: defs/gcs_uploader.py [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+      os.symlink(os.path.join(argv.root, artifact), os.path.join(scratch, artifact))
```
Comment (spxtr) [inline defs/gcs_uploader.py]: rmtree doesn't follow symlinks does it?

Comment (mikedanese) [inline defs/gcs_uploader.py]: nope.

File: defs/gcs_uploader.py (hunk 7 above)
Comment (mikedanese) [inline defs/gcs_uploader.py]: yup. added a comment.

**Comment by mikedanese** [comment]
@spxtr fixed.

File: defs/build.bzl [hunk 9]
```diff
@@ -0,0 +1,43 @@
+def _gcs_upload_impl(ctx):
+  targets = []
```
Comment (spxtr) [inline defs/build.bzl]: You mix two and four spaces in a few places.

File: defs/deb.bzl [hunk 10]
```diff
@@ -0,0 +1,34 @@
+load("@bazel_tools//tools/build_defs/pkg:pkg.bzl", "pkg_tar", "pkg_deb")
+
+KUBERNETES_AUTHORS = "Kubernetes Authors <kubernetes-dev+release@googlegroups.com>"
+
+KUBERNETES_HOMEPAGE = "https://kubernetes.io"
```
Comment (spxtr) [inline defs/deb.bzl]: Actually it's `http`, which is pretty gross.

File: defs/gcs_uploader.py [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+import atexit
+import os
+import os.path
+import shutil
+import subprocess
+import sys
+import tempfile
+
+
+parser = argparse.ArgumentParser(description='Upload build targets to GCS.')
```
Comment (spxtr) [inline defs/gcs_uploader.py]: Move this into the `if __name__ == '__main__':` block.

File: defs/gcs_uploader.py [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+            artifact = artifact.strip()
+            try:
+                os.makedirs(os.path.join(scratch, os.path.dirname(artifact)))
+            except (OSError):
+                # skip directory already exists errors
+                pass
+            os.symlink(os.path.join(argv.root, artifact), os.path.join(scratch, artifact))
+
+    sys.exit(subprocess.call(["gsutil", "-m", "rsync", "-C", "-r", scratch, argv.gcs_path]))
+
```
Comment (spxtr) [inline defs/gcs_uploader.py]: Two newlines between top-level objects.

File: defs/build.bzl (hunk 9 above)
Comment (mikedanese) [inline defs/build.bzl]: skylark style guide says 2 space for regular indentation, 4 spaces for hanging indentation

File: defs/gcs_uploader.py (hunk 12 above)
Comment (mikedanese) [inline defs/gcs_uploader.py]: Done.

**Comment by mikedanese** [comment]
@spxtr addressed comments

File: defs/gcs_uploader.py [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+
+if __name__ == '__main__':
+    main(parser.parse_args())
```
Comment (mikedanese) [inline defs/gcs_uploader.py]: this is broken

**Pull request merged**